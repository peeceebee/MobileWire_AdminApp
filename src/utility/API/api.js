import axios from 'axios';

export default axios.create({
  baseURL: 'http://34.233.181.46/MW3/'
});

