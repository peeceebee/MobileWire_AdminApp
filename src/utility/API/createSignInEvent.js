import API from "./api";

const createSignInEvent = async body => {
  const request = JSON.stringify(body);
  try {
    const response = await API.post(`api/signin/CreateSignInEvent`, request, {
      headers: { "Content-Type": "application/json" }
    });
    console.log(response); 
    return response.data;
  } catch (e) {
    return 'Error Creating Event'
  }
};

export default createSignInEvent;
