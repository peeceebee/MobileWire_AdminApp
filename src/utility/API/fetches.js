import API from "./api";

export const createSignInEvent = async body => {
  const request = JSON.stringify(body);
  try {
    const response = await API.post(`api/signin/CreateSignInEvent`, request, {
      headers: { "Content-Type": "application/json" }
    });
    return response.data;
  } catch (e) {
    return "Error Creating Event";
  }
};
export const signInEmployee = async body => {
  const request = JSON.stringify(body);
  try {
    const response = await API.post(`api/signin/SignInEmployee`, request, {
      headers: { "Content-Type": "application/json" }
    });
    console.log(response.data);
    return response.data;
  } catch (e) {
    console.log(e);
  }
};

export const signInNewEmployee = async body => {
  const request = JSON.stringify(body);
  try {
    const response = await API.post(`api/signin/SignInNewEmployee`, request, {
      headers: { "Content-Type": "application/json" }
    });
    console.log(response.data);
    return response.data;
  } catch (e) {
    return "Error creating employees";
  }
};
export const updateEmployee = async body => {
  const request = JSON.stringify(body);
  try {
    const response = await API.post(`api/settings/UpdateEmployee`, request, {
      headers: { "Content-Type": "application/json" }
    });
    return response.data;
  } catch (e) {
    return "Error updating employee!";
  }
};
export const updatePosition = async body => {
  const request = JSON.stringify(body);
  try {
    const response = await API.post(`api/settings/UpdatePosition`, request, {
      headers: { "Content-Type": "application/json" }
    });
    return response.data;
  } catch (e) {
    return "Error updating position!";
  }
};

export const getEmpsAndPositns = async () => {
  const res = await API.get("api/settings/GetEmpsAndPos");
  return res.data;
};

export const getEmployees = async () => {
  const res = await API.get("api/settings/GetEmployees");
  return res.data;
};

export const getPositions = async () => {
  const res = await API.get("api/settings/GetPositions");
  return res.data;
};

export const getHomeView = async () => {
  const res = await API.get("api/signin/GetHomeView");
  const events = res.data.events;
  const stDt = res.data.missionStart;
  const endDt = res.data.missionEnd;
  const homeView = { events: events, stDt, endDt };
  return homeView;
};
export const getSignInView = async eventid => {
  const res = await API.get("api/signin/GetEventView/", {
    params: { eventId: eventid }
  });
  const evId = res.data.signInEventId;
  const evDate = res.data.eventDate;
  const evType = res.data.eventType;
  const signins = res.data.signIns;
  const employees = res.data.allEmployees;
  const positions = res.data.allPositions;
  const SignInData = {
    signins: signins,
    employees: employees,
    positions: positions,
    eventid: evId,
    eventdate: evDate,
    eventtype: evType
  };
  return SignInData;
};
