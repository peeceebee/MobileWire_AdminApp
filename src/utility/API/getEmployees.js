import API from "./api";

export const getEmployees = async () => {
    const res = await API.get("api/settings/GetEmployees")
    return res.data;
}