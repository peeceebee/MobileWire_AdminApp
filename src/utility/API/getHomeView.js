import API from "./api";
const getHomeView = async () => {
  const res = await API.get("api/signin/GetHomeView");
  const events = res.data.events;
  const stDt = res.data.missionStart;
  const endDt = res.data.missionEnd;
  const homeView = { events: events, startDate: stDt, endDate: endDt };
  console.log(homeView); 
  return homeView;
};
export default getHomeView;
