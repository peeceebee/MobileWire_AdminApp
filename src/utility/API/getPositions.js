import API from "./api";

export const getPositions = async () => {
  const res = await API.get("api/settings/GetPositions");
  return res.data;
};
