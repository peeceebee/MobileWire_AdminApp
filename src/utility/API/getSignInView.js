import API from './api';

const getSignInView = async (eventid) => {
  const res = await API.get("api/signin/GetEventView/", { params: {eventId: eventid}});
  const evId = res.data.signInEventId;
  const evDate = res.data.eventDate;
  const evType = res.data.eventType;
  const signins = res.data.signIns;   
  const employees = res.data.allEmployees;
  const positions = res.data.allPositions;
  const SignInData = 
        { 
            signins: signins, 
            employees: employees,
            positions: positions,
            eventid: evId,
            eventdate: evDate,
            eventtype: evType
        };
  return SignInData;
};

export default getSignInView;
