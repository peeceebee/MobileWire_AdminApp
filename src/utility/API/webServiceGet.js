import API from './api';

const WebServiceGet = (url, params) => {
  return API.get(url, params)
    .then(function(response) {
      //    console.log(response, "ws response");
      return {
        viewdata: response.data,
        successful: true,
        error: null
      };
    })
    .catch(function(err) {
      //  console.log(err, "ws err");
      if (!(err.message === null) && err.message === "Network Error") {
        return {
          viewdata: null,
          successful: false,
          error: "Network connection is not available."
        };
      }
      if (err.response !== "undefined") {
        return {
          viewdata: null,
          successful: false,
          error: err.message
        };
      }
      return {
        viewdata: null,
        successful: false,
        error: "No internet connection or server is not available"
      };
    });
};

export default WebServiceGet;
