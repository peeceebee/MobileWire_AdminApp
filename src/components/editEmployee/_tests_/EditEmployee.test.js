import EditEmployees from "../EditEmployee";
import React from "react";
import { shallow } from "enzyme";
import { updateEmployee } from "../../../utility/API/fetches";

jest.mock("../../../utility/API/fetches", () => ({
  updateEmployee: jest.fn()
}));

describe("EditEmployees", () => {
  let wrapper;
  const mockProps = {
    state: {
      employee: {
        employeeId: 1,
        lastName: "Silver",
        firstName: "Max",
        cellPhone: "1234567",
        lastBarcode: "S10043P1",
        lastPosition: 1,
        isTerminated: false
      }
    }
  };
  beforeEach(() => {
    wrapper = shallow(<EditEmployees location={mockProps} />);
  });
  it("Should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Should have a default state equivalent to props", () => {
    expect(wrapper.state()).toEqual({
      barcode: "S10043P1",
      cellPhone: "1234567",
      employeeId: 1,
      error: "",
      firstName: "Max",
      lastName: "Silver",
      position: 1,
      terminated: false
    });
  });
  describe("handleChange", () => {
    beforeEach(() => {
      wrapper = shallow(<EditEmployees location={mockProps} />);
    });
    it("Should change state on firstName input change", () => {
      expect(wrapper.state("firstName")).toEqual("Max");
      wrapper
        .find({ name: "firstName" })
        .simulate("change", { target: { name: "firstName", value: "Peter" } });
      wrapper.update();
      expect(wrapper.state("firstName")).toEqual("Peter");
    });
    it("Should change state on lastName input change", () => {
      expect(wrapper.state("lastName")).toEqual("Silver");
      wrapper
        .find({ name: "lastName" })
        .simulate("change", { target: { name: "lastName", value: "Baxter" } });
      wrapper.update();
      expect(wrapper.state("lastName")).toEqual("Baxter");
    });
    it("Should change state on cellPhone input change", () => {
      wrapper.find({ name: "cellPhone" }).simulate("change", {
        target: { name: "cellPhone", value: "7654321" }
      });
      wrapper.update();
      expect(wrapper.state("cellPhone")).toEqual("7654321");
    });
    it("Should change state on barcode input change", () => {
      expect(wrapper.state("barcode")).toEqual("S10043P1");
      wrapper
        .find({ name: "barcode" })
        .simulate("change", { target: { name: "barcode", value: "S10043P5" } });
      wrapper.update();
      expect(wrapper.state("barcode")).toEqual("S10043P5");
    });
  });
  describe("handleToggle", () => {
    it("Should toggle the terminated boolean when handleToggle is called", () => {
      expect(wrapper.state("terminated")).toEqual(false);
      wrapper.find({ name: "terminated" }).simulate("click");
      wrapper.update();
      setTimeout(() => {
        expect(wrapper.state("terminated")).toEqual(true);
      }, 1000);
    });
  });
  describe("handleSaveEmployee", () => {
    it("Should call updateEmployee when the save button is clicked", () => {
      wrapper.find(".save-btn").simulate("click", { preventDefault: () => {} });
      expect(updateEmployee).toHaveBeenCalled();
    });
    it.skip("Should call updateEmployee when the save button is clicked", () => {
      updateEmployee.mockImplementation(() => Promise.reject());
      wrapper.find(".save-btn").simulate("click", { preventDefault: () => {} });
      expect(wrapper.state("error")).toEqual(null);
    });
  });
});
