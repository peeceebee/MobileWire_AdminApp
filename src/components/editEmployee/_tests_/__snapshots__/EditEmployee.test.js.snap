// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditEmployees Should match the snapshot 1`] = `ShallowWrapper {}`;
